File: Copy_String_Remove_Extra_WhiteSpace/Copy.py
"""
    Author : Pakshal Shshikant Jain
    Date : 30/09/2023
    Program : Write a program which accept string from user and copy the
              contents into another string by removing extra white spaces.
                
              Input : Marvellous        multi          OS”
              Output : Marvellous multi OS”

"""

#########################################################################################################################################

#Class Declaration
class StringCopy :
    #Characateristics of Class
    str = " ";
    arr = [];
    brr = [];

    #Constructor of Class
    def __init__(self) :
        self.str = " ";
        self.arr = [];
    
    #Behaviour of Class To Take Input from user
    def Accept(self) :
        #Taking input from user
        print("Enter One String");
        self.str = input();

        self.arr = list(self.str);
    
    #Behaviour of Class To Copy String After Removing White Spaces
    def StrCpyX(self) :
        i = 0;

        if((self.brr == None)and(self.brr == None)) :
            return;
        
        #Logic To Remove
        while(i != len(self.arr)) :
            if(self.arr[i] == ' ') :
                while((self.arr[i] == ' ')and(i != len(self.arr))) :
                    i = i + 1;
                self.brr.append(' ');
            
            self.brr.append(self.arr[i]);
            i = i + 1;

#######################################################################################################################################

#Main Function
def main() :
    print("Jay Ganesh....");

    #Creating object of StringCopy Class
    sobj = StringCopy();

    #Call To Behaviours of Class
    sobj.Accept();
    sobj.StrCpyX();

    #Printing Copied String
    print("Copied String After Removing White Spaces is : ","".join(sobj.brr));

################################################################################################################################################

#Call Back Method To Call Main Function
if __name__ == "__main__" :
    #Call To Main Function
    main();

File: Copy_String_Remove_Extra_WhiteSpace/Copy.js
/*
    Author : Pakshal Shshikant Jain
    Date : 30/09/2023
    Program : Write a program which accept string from user and copy the
              contents into another string by removing extra white spaces.
                
              Input : Marvellous        multi          OS”
              Output : Marvellous multi OS”

*/
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//Class Declaration
var StringCopy = /** @class */ (function () {
    //Constructor of Class
    function StringCopy() {
        this.str = " ";
        this.arr = [];
        this.brr = [];
    }
    //Behaviour of Class To Take input from user
    StringCopy.prototype.Accept = function () {
        this.str = "Pakshal            Jain";
        this.arr = Array.from(this.str);
    };
    //Behaviour of Class To Copy String After Removing White Spaces
    StringCopy.prototype.StrCpyX = function () {
        var i = 0;
        var j = 0;
        if ((this.arr == null) && (this.brr == null)) {
            return;
        }
        //Logic To Copy String
        while (i != this.arr.length) {
            if (this.arr[i] == ' ') {
                while ((this.arr[i] == ' ') && (i != this.arr.length)) {
                    i++;
                }
                this.brr[j] = ' ';
                j++;
            }
            this.brr[j] = this.arr[i];
            j++;
            i++;
        }
    };
    return StringCopy;
}()); //End of Class
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//Main Function
function main() {
    console.log("Jay Ganesh....");
    //Creating object of StringCpoy Class
    var sobj = new StringCopy();
    //Call To Behaviours of Class
    sobj.Accept();
    sobj.StrCpyX();
    //Printing Copied String
    console.log("Copied String is :", sobj.brr.join(""));
}
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//Call To Main Function
main();
